webrender/src/device/gfx/device.rs

```diff
@@ -1152,23 +1281,85 @@ impl<B: hal::Backend> Device<B> {
         }
     }
 
+    pub fn set_uniforms(&mut self, program: &ProgramId, projection: &Transform3D<f32>) {
+        debug_assert!(self.inside_frame);
+        assert_ne!(self.bound_program, INVALID_PROGRAM_ID);
+        assert_eq!(*program, self.bound_program);
+
+        let locals = Locals {
+            uTransform: projection.to_row_arrays(),
+            uMode: self.program_mode_id,
+        };
+
+        let bound_locals = if self.locals.contains_key(&locals) {
```

Comment:
let's use the `Entry` API here, i.e. `let bound_locals = match self.locals.entry(locals) {..}`